HOM project

PMehrfeld opened the pull request:
Finally, after appr. 1 year of development, this branch can be merged into development.  
This is related to the project [HighOrderModel](https://github.com/RWTH-EBC/AixLib/projects/5)

The main changes are:
- Partial room model
- Much better propagation from wall level to building level for OFD rooms and building
- Comparative validation according to ASHRAE
- Empirical validation on the basis of two measurement references

Many variables do not meet the naming conventions yet. However, this was not the aim. The aim was to boost the HOM approach from perspectives like usability, contents, error-proneness

FWuellhorst commented on AixLib/Resources/Scripts/ConvertAixLib_from_0.11.2_to_0.11.3.mos:
```diff
@@ -0,0 +1,13 @@
+convertClass("AixLib.ThermalZones.HighOrder.Rooms.MFD.CellarAttic.Cellar", "AixLib.Obsolete.Year2020.ThermalZones.HighOrder.Rooms.MFD.CellarAttic.Cellar");
+convertElement("AixLib.ThermalZones.HighOrder.Rooms.BaseClasses.PartialRoomParams", "calcMethod", "calcMethodOut");
+convertElement("AixLib.ThermalZones.HighOrder.Components.Walls.Wall", "calcMethod", "calcMethodOut");
+convertModifiers("AixLib.ThermalZones.HighOrder.Components.Walls.Wall", "solarDistribution",  fill("",0));
```
I get the following error:
```
Argument number 2, oldElements = "solarDistribution", in
convertModifiers("AixLib.ThermalZones.HighOrder.Components.Walls.Wall", "solarDistribution", fill("", 0), false)
must be a vector.
```

PMehrfeld replied:
Solved in 8ea49cd054a2b167f2e10c1ad7e0437ec49774c7

FWuellhorst commented on AixLib/package.mo:
```diff
@@ -35,10 +35,13 @@ package AixLib
+    version="0.10.4", script="modelica://AixLib/Resources/Scripts/ConvertAixLib_from_0.10.4_to_0.10.5.mos",
+    version="0.10.5", script="modelica://AixLib/Resources/Scripts/ConvertAixLib_from_0.10.5_to_0.10.6.mos",
+    version="0.10.6", script="modelica://AixLib/Resources/Scripts/ConvertAixLib_from_0.10.6_to_0.10.7.mos",
+    version="0.11.0", script="modelica://AixLib/Resources/Scripts/ConvertAixLib_from_0.11.0_to_0.11.1.mos",
+    version="0.11.1", script="modelica://AixLib/Resources/Scripts/ConvertAixLib_from_0.11.1_to_0.11.2.mos",
+    version="0.11.2", script="modelica://AixLib/Resources/Scripts/ConvertAixLib_from_0.11.2_to_0.11.3.mos")),
```
Why increase the numbers three times in comparison to dev?

PMehrfeld replied:
Your are right. Was due to various sub-branches within the HOM project. Resolved in 201422ebd4e6644a8ec582a7c315a5a8d864b5fc

FWuellhorst commented on AixLib/ThermalZones/HighOrder/House/MFD/BuildingEnvelope/OneAppartment_VoWo.mo:
```diff
@@ -20,32 +20,51 @@ model OneAppartment_VoWo
   parameter Modelica.SIunits.Temperature TOutAirLimit
     "Temperature at which sunblind closes (see also solIrrThreshold)"
     annotation(Dialog(group = "Sunblind", enable=use_sunblind));
+
+  replaceable model WindowModel = AixLib.ThermalZones.HighOrder.Components.WindowsDoors.BaseClasses.PartialWindow
+    constrainedby AixLib.ThermalZones.HighOrder.Components.WindowsDoors.BaseClasses.PartialWindow annotation (Dialog(tab="Outer walls", group="Windows"), choicesAllMatching = true);
+
+  replaceable model CorrSolarGainWin = AixLib.ThermalZones.HighOrder.Components.WindowsDoors.BaseClasses.CorrectionSolarGain.PartialCorG
```
All models in `AixLib.ThermalZones.HighOrder.Rooms.MFD` don't use the partial room model! Is this on purpose?

PMehrfeld replied:
Absolutely! I admit that I wanted to do this still in this branch. However, I already raised an issue #978. So we can solve this after this PR from `development` as basis.

FWuellhorst commented on AixLib/ThermalZones/HighOrder/Rooms/BaseClasses/PartialRoom.mo:
```diff
@@ -0,0 +1,97 @@
+within AixLib.ThermalZones.HighOrder.Rooms.BaseClasses;
+partial model PartialRoom "Partial model with base component that are necessary for all HOM rooms"
+
+  extends PartialRoomParams;
+
+  // Air volume of room
+  parameter Modelica.SIunits.Volume room_V annotation (Dialog(group="Air volume of room"));
+
```
I was wondering if another `PartialRoom`, `PartialRoomFourWalls` would have been an option as all examples have at least 4 walls, a floor and a ceiling. 
This would also make sense as the record already contains the information on these wall types.
However, this might be something for future developments.

PMehrfeld replied:
I agree. However, having such a base class with ceiling, floor and 4 walls can still be added in a further issue.

FWuellhorst reviewed (changes requested):
Nice work! I haven't looked at all Resources and HOM files as it is just too much and you did internal reviews beforehand.
Besides the conversion error and some open questions, I have no requests.

PMehrfeld commented on AixLib/Fluid/Actuators/package.mo:
```diff
@@ -32,14 +32,50 @@ package UsersGuide "User's Guide"
 
 
 
+
+
+
+
+
+
```
Solved in 219b06181c81a537470463072a783f68d1139cf7

PMehrfeld commented on AixLib/Obsolete/Year2020/ThermalZones/HighOrder/Rooms/MFD/CellarAttic/Attic_Ro2Lf1.mo:
```diff
@@ -1,7 +1,9 @@
-within AixLib.ThermalZones.HighOrder.Rooms.MFD.CellarAttic;
+within AixLib.Obsolete.Year2020.ThermalZones.HighOrder.Rooms.MFD.CellarAttic;
```
Only the classes for cellar and attic of the MFH were moved, since they are not in use in any direct AixLib example.

PMehrfeld commented on AixLib/Fluid/Examples/GeothermalHeatPump/BaseClasses/GeothermalHeatPumpBase.mo:
```diff
@@ -16,16 +16,12 @@ partial model GeothermalHeatPumpBase
-        9700,11600,13000,14800,16300; 323.15,10000,11200,12900,16700,
-        17500]) "Base load energy conversion unit"
-    annotation (Placement(transformation(extent={{-40,-14},{-4,20}})));
+  AixLib.Obsolete.Year2019.Fluid.HeatPumps.HeatPumpSimple heatPumpTab(
+    volumeEvaporator(T_start=T_start_cold[1]),
+    volumeCondenser(T_start=T_start_warm[5]),
+    redeclare package Medium = Medium,
+    tablePower=[0,266.15,275.15,280.15,283.15,293.15; 308.15,3300,3400,3500,3700,3800; 323.15,4500,4400,4600,5000,5100],
+    tableHeatFlowCondenser=[0,266.16,275.15,280.15,283.15,293.15; 308.15,9700,11600,13000,14800,16300; 323.15,10000,11200,12900,16700,17500]) "Base load energy conversion unit" annotation (Placement(transformation(extent={{-40,-14},{-4,20}})));
```
Will be solved in #1093.

PMehrfeld commented on AixLib/DataBase/Walls/Collections/EmpricalValidation/Building1.mo:
```diff
@@ -0,0 +1,17 @@
+within AixLib.DataBase.Walls.Collections.EmpricalValidation;
+record Building1
```
(Hopefully) solved in 42109219567f1722f8a35decaebed91ec48892c0. Needs to be checked if CI finishes successfully.

PMehrfeld replied:
Due to historic development of this example of emp. validation, the use case was still named Building1. However, Warehouse is the new and more precise use-case name now.

PMehrfeld commented on AixLib/DataBase/ThermalZones/ZoneRecordDummy.mo:
```diff
@@ -72,6 +72,8 @@ record ZoneRecordDummy "This is a dummy record with non-physical parameter value
     withAHU=false,
     minAHU=Modelica.Constants.eps,
     maxAHU=2*Modelica.Constants.eps,
+    shadingFactor=fill(1, nOrientations),
+    maxIrr=fill(0, nOrientations),
```
To be honest, I am not sure. I assume one of the contributors added it due to a check error.  
I have noticed that it is included in the `development` branch in the `ZoneBaseRecord` ([link to line](https://github.com/RWTH-EBC/AixLib/blob/development/AixLib/DataBase/ThermalZones/ZoneBaseRecord.mo#L101)) but not in the `ZoneRecordDummy` ([link to line](https://github.com/RWTH-EBC/AixLib/blob/development/AixLib/DataBase/ThermalZones/ZoneRecordDummy.mo#L74)). 

Thus, I think this is a good addition and I would just leave it here in this PR.

PMehrfeld commented on AixLib/DataBase/WindowsDoors/Simple/WindowSimple_EnEV2009.mo:
```diff
@@ -1,6 +1,6 @@
 within AixLib.DataBase.WindowsDoors.Simple;
 record WindowSimple_EnEV2009 "Window according to EnEV 2009"
-  extends OWBaseDataDefinition_Simple(Uw = 1.3, g = 0.6, Emissivity = 0.9, frameFraction = 0.2);
```
It has not directly to do with the capacity. But yes, the simple window, does not have an emissivity factor for shortwave radiation (see `AixLib.ThermalZones.HighOrder.Components.WindowsDoors.WindowSimple`). Instead, it includes a correction model (`AixLib.ThermalZones.HighOrder.Components.WindowsDoors.BaseClasses.CorrectionSolarGain.PartialCorG`) which transforms (solar) shortwave radiation into a heat flow for the node that represents the mean (longwave) radiant temperature. Therefore, we deleted the factor `Emissivity` from the base record `AixLib.DataBase.WindowsDoors.Simple.OWBaseDataDefinition_Simple`.

PMehrfeld commented:
@FWuellhorst: I addressed all your comments. Thanks a lot! They were quite good and helpful. Thus, I re-requested your review. 

If the CI pipelines finish successfully and you approve the review, you (or I) can merge and delete the branch `HOM_project`.

FWuellhorst commented on AixLib/Resources/Scripts/ConvertAixLib_from_0.10.5_to_0.11.0.mos:
```diff
@@ -0,0 +1,5 @@
+clear
+
+convertClear();
+
+convertClass("AixLib.ThermalZones.HighOrder.Components.WindowsDoors.BaseClasses.CorrectionSolarGain.NoCorG", "AixLib.ThermalZones.HighOrder.Components.WindowsDoors.BaseClasses.CorrectionSolarGain.CorGSimple")
```
One last thing, shouldn't this conversion script go from 0.10.5 to 0.10.6 or 0.10.7?

PMehrfeld replied:
Good eye ;) Solved in eafec7023aba1c8f648ac7a42b9a8c85a60ae734

FWuellhorst reviewed (approved):
Nice work! Besides my last question, everything looks fine to me :) As you know more about these scripts, I will approve anyways. You can check if it needs a fix. 

I will also leave you the honours to merge and delete the branch ;)

FWuellhorst reviewed (approved):
Everything looks good!
